crypto/src/hash/rpo/digest.rs

```diff
@@ -0,0 +1,112 @@
+use super::DIGEST_SIZE;
+use crate::{Digest, Felt, StarkField};
+use core::slice;
+use winterfell::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
+
+// DIGEST TRAIT IMPLEMENTATIONS
+// ================================================================================================
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct RpoDigest256([Felt; DIGEST_SIZE]);
+
+impl RpoDigest256 {
+    pub fn new(value: [Felt; DIGEST_SIZE]) -> Self {
+        Self(value)
+    }
+
+    pub fn as_elements(&self) -> &[Felt] {
+        &self.0
+    }
```

Comment:
I like this! One question: what's the reason for using `Deref` rather than `AsRef`?

Also, we should probably import from `core::ops` rather than from `std::ops`.

Reply:
`Deref` will extend all the implementations of `&Self::Target`. Example: `String` implements `Deref<Target=&str>`, so it will provide all methods implemented only for `&str`, such as [str::chars](https://doc.rust-lang.org/std/string/struct.String.html#method.chars)

The problem is you can have only one `Deref` per type. `AsRef` is useful when you want the same type to have references to multiple distinct types.

As for the `core::ops`, indeed we should go that way so we keep compatibility with `no-std`

Reply:
Thanks, I have implemented the `Deref` trait as suggested.